v6-t-test-py/partial.py

```diff
@@ -40,28 +43,40 @@ def partial(df: pd.DataFrame, column_name: str) -> dict:
             f"{MINIMUM_NUMBER_OF_RECORDS}."
         )
 
-    if not ptypes.is_numeric_dtype(df[column_name]):
-        error("Column must be numeric.")
+    if not columns:
+        columns = df.select_dtypes(include=["int", "float"]).columns.tolist()
+    else:
+        # Check that column names exist in the dataframe
+        if not all([col in df.columns for col in columns]):
+            non_existing_columns = [col for col in columns if col not in df.columns]
+            raise InputError(
+                f"Columns {non_existing_columns} do not exist in the dataframe"
+            )
```

Comment:
It could be slightly more readable:
```suggestion
        non_existing_columns = [col for col in columns if col not in df.columns]
        if non_existing_columns:
            raise InputError(
                f"Columns {non_existing_columns} do not exist in the dataframe"
            )
```